### PR TITLE
Worked on the Admin Intent Testing. Resolved the big related to async…

### PR DESCRIPTION
**Pull request by Shubjaura**
Developed BaseIntentTest, a base class that standardizes our Espresso tests.
Added methods for retrying interactions with UI elements, allowing a specified number of attempts with delays. These methods handle: Actions like clicks on views identified either by ID or text, and Checks for view assertions, again based on either view IDs or text.
Developed thorough and comprehensive test cases focused on the administrative features of the application.
Ensured robust validation of intent functionalities specific to admin operations along with attendee profile creations.
Corrected the event intent testing code based on the new UI implementation
Lastly, added Javadocs.

### CODE REVIEW

**Review by Aryan-Rusia-2 — APPROVED** [review]
Looks good 💯

File: code/EventLinkQR/app/src/main/res/layout/create_event_page.xml [hunk 1]
```diff
@@ -199,6 +184,21 @@
                 android:textSize="20sp"
                 android:textStyle="bold" />
 
+            <TextView
```
Comment (adrienz1) [inline code/EventLinkQR/app/src/main/res/layout/create_event_page.xml:187]: why did you move this block

**Review by adrienz1 — APPROVED** [review]
why did you move

File: code/EventLinkQR/app/src/main/res/layout/create_event_page.xml (hunk 1 above)
Comment (Shubjaura) [reply code/EventLinkQR/app/src/main/res/layout/create_event_page.xml:187]: In the intent testing, it is essential to have the components displayed on the screen that we are testing, so to keep the intent testing simple without having to scroll the page, I moved geolocation toggle button up (so that it is displayed on the screen while testing)
